### PR TITLE
Deserialisation performance enhancements

### PR DESCRIPTION
**Pull request by mvkra**
Some more code to improve the deserialisation side this time.  A synthetic benchmark reduces the performance gains I experienced in my real work app but still significant. 

Also added a write performance enhancement that came up when profiling: Replacing a "sprintf %A value" for the field-less union case to derive the union case name with just getting the case name previously resolved already also improves write performance as the "%A" reflection is no longer required.

A quick benchmark on my laptop for deserialisation (approx 20% increase in performance):
Original results [N: 1000000, TimeInMilliseconds: 125906]
New results [N: 1000000, TimeInMilliseconds: 100170].

### CODE REVIEW

File: FSharpLu.Json/Compact.fs [hunk 1]
```diff
@@ -28,6 +28,20 @@ module private ConverterHelpers =
     
     let getUnionCasesMemorised = memorise FSharpType.GetUnionCases
 
+    let constructUnionCaseMemorised = memorise FSharpValue.PreComputeUnionConstructor
```
Comment (blumu) [inline FSharpLu.Json/Compact.fs]: Naming suggestion: I would consider moving all the `memorized` let bindings under a `Memorised` submodule as follows:
```fs
module Memorised =
   let constructUntionCase = …
   let getUnionCase = …
```

File: FSharpLu.Json/Compact.fs [hunk 2]
```diff
@@ -28,6 +28,20 @@ module private ConverterHelpers =
...
+
+    let getUnionCaseProperyInfoFieldsMemorised = memorise (fun (case: UnionCaseInfo) -> case.GetFields())
+
+    let findNoFieldsMatchingUnionCaseByNameAndType  =
```
Comment (blumu) [inline FSharpLu.Json/Compact.fs]: Naming consistency: `findNoFieldsMatchingUnionCaseByNameAndType` does not end with `Memorised` like the two let bindings above. I recommend moving everything under `Memorised` module and removing the suffix everywhere.

**Comment by mvkra** [comment]
I have no merge access for this PR. If/when you are ready you can merge and release the code.

**Comment by mvkra** [comment]
Is it possible to get a nuget package released with the changes? I note that the last Nuget release was before this change.

**Comment by blumu** [comment]
@mvkra I've just released 0.10.28 on nuget.

**Pull request merged**